comp/core/config/config.go

```diff
@@ -25,6 +26,10 @@ type dependencies struct {
 }
 
 func newConfig(deps dependencies) (Component, error) {
+	if comptest.IsTest {
+		panic("do not use non-mock comp/core/config in tests")
```

Comment:
`panic` feels an OK thing to do here.... But was also considering:
```golang
return nil, error.New("do not use non-mock comp/core/config in tests")
```

Reply:
Yeah, that's probably better.  I like to use panics for programming errors, especially when there's not an easily accessible error return that will crash the agent.  But in this case, such a thing is available!